acm/registration/odh-core/odh-core.application.yaml

```
@@ -0,0 +1,73 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: odh-core-acm
+---
+apiVersion: app.k8s.io/v1beta1
+kind: Application
+metadata:
+  name: odh-core
+  namespace: odh-core-acm
+spec:
+  componentKinds:
+  - group: apps.open-cluster-management.io
+    kind: Subscription
+  descriptor: {}
+  selector:
+    matchExpressions:
+      - key: app
+        operator: In
+        values: 
+          - odh-core
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: Channel
+metadata:
+  annotations:
+    apps.open-cluster-management.io/reconcile-rate: medium
+  name: channel
+  namespace: odh-core-acm
+spec:
+  type: Git
+  pathname: 'https://github.com/opendatahub-io/ai-edge'
+---
+apiVersion: cluster.open-cluster-management.io/v1beta2
+kind: ManagedClusterSet
+metadata:
+  name: odh-core
+spec:
+  clusterSelector:
+    selectorType: ExclusiveClusterSetLabel
```

I think the ManagedClusterSetBinding will also be needed here:

```suggestion
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSet
metadata:
  name: odh-core
spec:
  clusterSelector:
    selectorType: ExclusiveClusterSetLabel
---
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSetBinding
metadata:
  name: odh-core
  namespace: odh-core-acm
spec:
  clusterSet: odh-core
```

^with the potential for some `kustomize` replacements akin to the ones I suggested earlier, to make sure that `spec.clusterSet` here refers to the name of the `ManagedClusterSet` above it.

---

Thinking about this a little more, and playing with the equivalent near-edge stuff, I'm starting to think that maybe the `ManagedClusterSet` _shouldn't_ be specified here, just the `ManagedClusterSetBinding`.

Reasoning: the rest of this is the resources to configure an application in ACM, whereas the `ManagedClusterSet` is more the infrastructure that can be common to many applications. This is emphasized by the fact that `ManagedClusterSet` is a cluster-scoped CRD, while all of the rest of these are namespace-scoped.

Keeping the `ManagedClusterSet` here would start to get weird with `kustomize`, if we do things like create multiple applications (which seems like a normal case) based off a common base, and use things like `namePrefix`: we'd end up with multiple `ManagedClusterSets`, where each one has just one application. We'd then have to make sure that the `ManagedClusters` are all labelled correctly for all of them.

In short: `ManagedClusterSet` should be defined alongside `ManagedClusters`, and `ManagedClusterSetBinding` should be here with the application resources. WDYT?